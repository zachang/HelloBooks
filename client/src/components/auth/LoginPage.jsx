import React from 'react';
import { Link } from 'react-router';
import HomepageHeader from '../common/HomepageHeader';
import LoginForm from './LoginForm';

export default class LoginPage extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      showLoginText: false
    };
  }

  render() {
    return (
      <div className="row main-row main-div">
        <HomepageHeader btnText={this.state.showLoginText}/>

        <div className="section no-pad-bot" id="index-banner">
          <div className="container textContent">
            <div className="row">

              <div className="col s8 offset-s2 sloginContent">
                <br/><br/>
                <LoginForm/>
                <h6 className="infoText">No account yet...?<Link to="/Register">Register</Link></h6>
                <br/><br/>
              </div>

            </div>
          </div>

        </div>
      </div>

    );
  }
}