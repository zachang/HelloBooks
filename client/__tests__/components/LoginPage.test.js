import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import LoginPage from '../../src/components/auth/LoginPage.jsx';

describe('<LoginPage/>', () => {
  const props = {
    showLoginText: false
  };

  const shallowWrapper = shallow(<LoginPage {...props}/>);
  const tree = toJson(shallowWrapper);

  it('renders <LoginPage /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it(`should check if the 'HomepageHeader' is defined`, () => {
    expect(shallowWrapper.find('HomepageHeader').length).toBe(1);
  });

  it(`should check if the 'LoginForm' is defined`, () => {
    expect(shallowWrapper.find('Connect(LoginForm)').length).toBe(1);
  });
});

