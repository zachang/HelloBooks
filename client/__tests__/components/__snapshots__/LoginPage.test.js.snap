// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage/> renders <LoginPage /> component 1`] = `
<div
  className="row main-row main-div"
>
  <HomepageHeader
    btnText={false}
  />
  <div
    className="section no-pad-bot"
    id="index-banner"
  >
    <div
      className="container textContent"
    >
      <div
        className="row"
      >
        <div
          className="col s8 offset-s2 sloginContent"
        >
          <br />
          <br />
          <Connect(LoginForm) />
          <h6
            className="infoText"
          >
            No account yet...?
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/Register"
            >
              Register
            </Link>
          </h6>
          <br />
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
